crates/bitcoind_rpc/src/lib.rs

```diff
@@ -84,24 +105,49 @@ impl<'c, C: bitcoincore_rpc::RpcApi> Emitter<'c, C> {
             // `start_height` has been emitted.
             .unwrap_or(self.start_height.saturating_sub(1));
 
+        // Get the raw mempool result from the RPC client which will be used to determine if any
+        // transactions have been evicted.
+        let raw_mempool = client.get_raw_mempool_verbose()?;
+        let raw_mempool_txids: HashSet<Txid> = raw_mempool.keys().copied().collect();
+
+        // Determine if height and hash matches the best block from the RPC. Evictions are deferred
+        // if we are not at the best block.
+        let height = client.get_block_count()?;
+        let at_tip = if height != self.last_cp.height() as u64 {
+            false
+        } else {
+            // Verify if block hash matches in case of re-org.
+            client.get_block_hash(height)? == self.last_cp.hash()
+        };
+
+        // If at tip, any expected txid missing from raw mempool is considered evicted;
+        // if not at tip, we don't evict anything.
+        let mut evicted_txids: HashSet<Txid> = if at_tip {
+            self.expected_mempool_txids
+                .difference(&raw_mempool_txids)
+                .copied()
+                .collect()
+        } else {
+            HashSet::new()
+        };
+
         // Mempool txs come with a timestamp of when the tx is introduced to the mempool. We keep
         // track of the latest mempool tx's timestamp to determine whether we have seen a tx
         // before. `prev_mempool_time` is the previous timestamp and `last_time` records what will
         // be the new latest timestamp.
         let prev_mempool_time = self.last_mempool_time;
         let mut latest_time = prev_mempool_time;
 
-        let txs_to_emit = client
-            .get_raw_mempool_verbose()?
+        let new_txs = raw_mempool
             .into_iter()
             .filter_map({
                 let latest_time = &mut latest_time;
+                let evicted_txids = &mut evicted_txids;
                 move |(txid, tx_entry)| -> Option<Result<_, bitcoincore_rpc::Error>> {
                     let tx_time = tx_entry.time as usize;
                     if tx_time > *latest_time {
                         *latest_time = tx_time;
                     }
-
                     // Avoid emitting transactions that are already emitted if we can guarantee
```

Comment:
Could you help me replace this code comment with this?

```rust
                    // Best-effort check to avoid re-emitting transactions we've already emitted.
                    //
                    // Complete suppression isn't possible, since a transaction may spend outputs
                    // owned by the wallet. To determine if such a transaction is relevant, we must
                    // have already seen its ancestor(s) that contain the spent prevouts.
                    //
                    // Fortunately, bitcoind provides the block height at which the transaction
                    // entered the mempool. If we've already emitted that block height, we can
                    // reasonably assume the receiver has seen all ancestor transactions.
```